src/unread/unreadSelectors.js

```diff
@@ -69,7 +69,11 @@ export const getUnreadByPms: Selector<{| [number]: number |}> = createSelector(
  */
 export const getUnreadPmsTotal: Selector<number> = createSelector(
   getUnreadPms,
-  unreadPms => unreadPms.reduce((total, pm) => total + pm.unread_message_ids.length, 0),
+  getMutedUsers,
+  (unreadPms, mutedUsers) =>
+    unreadPms
+      .filter(pm => !mutedUsers.has(pm.sender_id))
+      .reduce((total, pm) => total + pm.unread_message_ids.length, 0),
```

Comment:
This logic isn't necessary -- the server guarantees that messages from muted users will never be unread. See API docs:
https://chat.zulip.org/api/mute-user

Reply:
Well, we've got a server bug, then :)

I think even once we fix that, it's prudent to keep this code in case of similar future problems.

Reply:
Started a discussion on [#backend > muted users](https://chat.zulip.org/#narrow/stream/3-backend/topic/muted.20users) on CZO for the backend bug part of this.

Reply:
Interesting, and it looks like the outcome of that thread is that this is an existing mobile bug! We're not maintaining the unread state correctly.

Specifically, when we handle `EVENT_NEW_MESSAGE` (i.e. the Redux action we make from a `message` event) in each of the `unread` sub-reducers, we have logic like this:
```js
      if (message.sender_id === getOwnUserId(globalState)) {
        return state;
      }
```

and instead it should be checking something like `message.flags.includes('read')`. So we should fix that.

Then I don't think it makes sense to also have logic here specific to muted users. The job of this bit of code is to count up part of the unreads data; if we have future bugs where the unreads data is wrong, we should fix the bugs by making the unreads data correct.

(If for example we did have a server bug in the future that gave us incorrect unreads data, the place I'd want to put a workaround like this is at the input end, to make the data correct in Redux. One reason is that that makes sure it's fixed consistently for all the different places we might be using that data.)